src/dxtbx/nexus/nxmx.py

```diff
@@ -406,20 +411,21 @@ def offset(self) -> pint.Quantity | None:
         example, as the mechanical offset from mounting the axis to its dependency.
         """
         if "offset" in self._handle.attrs:
-            return self._handle.attrs.get("offset") * ureg(self.offset_units or "")
+            return self._handle.attrs["offset"] * self.offset_units
         return None
 
     @cached_property
-    def offset_units(self) -> str | None:
+    def offset_units(self) -> pint.Unit:
         """Units of the offset. Values should be consistent with NX_LENGTH."""
         if "offset_units" in self._handle.attrs:
-            return h5str(self._handle.attrs.get("offset_units"))
+            return ureg.Unit(h5str(self._handle.attrs["offset_units"]))
         # This shouldn't be the case, but DLS EIGER NeXus files include offset without
         # accompanying offset_units, so use units instead (which should strictly only
         # apply to vector, not offset.
         # See also https://jira.diamond.ac.uk/browse/MXGDA-3668
         # logger.warning(
-        # f"'offset_units' attribute not present for {self.path}, falling back to 'units'"
+        # f"'offset_units' attribute not present for {self.path}, "
```

Comment:
We can probably remove this commented code?

Reply:
Sure, probably, I didn't know what it was for so I didn't want to go in with a chainsaw, but I agree it looks like a fossil.

Regarding the walruses, I've only used them where an `h5py.Dataset` (or `None`) is expected, so see comment below (tl;dr: this is OK, but could do with a test).